evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -81,8 +81,14 @@ class DropColumns(ColumnSelector):
     """{}"""
     needs_fitting = False
 
+    def _check_input_for_columns(self, X):
```

Comment:
Needed the same changes that I made to `SelectColumns` in #2944 due to the split in pipeline structure.